src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -194,35 +221,58 @@ def remap(self: _BaseNdArrayField, connectivity) -> _BaseNdArrayField:
 
     __pow__ = _make_binary_array_field_intrinsic_func("pow", "power")
 
-    def __getitem__(self, index: common.FieldSlice) -> common.Field | core_defs.ScalarT:
-        if (
-            not isinstance(index, tuple)
-            and not common.is_domain_slice(index)
-            or common.is_named_index(index)
-            or common.is_named_range(index)
-        ):
-            index = cast(common.FieldSlice, (index,))
+    __mod__ = __rmod__ = _make_binary_array_field_intrinsic_func("mod", "mod")
+
+    def __and__(self, other: common.Field) -> _BaseNdArrayField:
+        if self.dtype == core_defs.BoolDType():
+            return _make_binary_array_field_intrinsic_func("logical_and", "logical_and")(
+                self, other
+            )
+        raise NotImplementedError("`__and__` not implemented for non-`bool` fields.")
+
+    __rand__ = __and__
+
+    def __or__(self, other: common.Field) -> _BaseNdArrayField:
+        if self.dtype == core_defs.BoolDType():
+            return _make_binary_array_field_intrinsic_func("logical_or", "logical_or")(self, other)
+        raise NotImplementedError("`__or__` not implemented for non-`bool` fields.")
+
+    __ror__ = __or__
+
+    def __xor__(self, other: common.Field) -> _BaseNdArrayField:
+        if self.dtype == core_defs.BoolDType():
+            return _make_binary_array_field_intrinsic_func("logical_xor", "logical_xor")(
+                self, other
+            )
+        raise NotImplementedError("`__xor__` not implemented for non-`bool` fields.")
+
+    __rxor__ = __xor__
+
+    def __invert__(self) -> _BaseNdArrayField:
+        if self.dtype == core_defs.BoolDType():
+            return _make_unary_array_field_intrinsic_func("invert", "invert")(self)
+        raise NotImplementedError("`__invert__` not implemented for non-`bool` fields.")
+
+    def _slice(self, index: common.FieldSlice) -> tuple[common.Domain, common.BufferSlice]:
```

Comment:
It may be useful to have `_slice` as a utility method so that it can be used by other fields such as `ConstantField` as the slicing methodology is shared.

Reply:
done